hugging_stanza.py

```diff
@@ -82,52 +66,32 @@ def push_to_hub():
         # Create the repository
         repo_name = "stanza-" + model
         repo_id = "stanfordnlp/" + repo_name
-        repo_url = api.create_repo(
-            repo_id=repo_id,
-            token=HfFolder.get_token(),
-            exist_ok=True,
-        )
-
-        # Clone the repository
-        repo_local_path = os.path.join("hub", repo_name)
-
-        repo = Repository(repo_local_path, clone_from=repo_url)
-        # checkout "main" so that we know we are tracking files correctly
-        repo.git_checkout("main")
-        if not repo.is_repo_clean():
-            print(f"Repo {repo_local_path} is currently not clean.  Unwilling to proceed...")
-            break
-        repo.git_pull(rebase=True)
-
-        # Make sure jar files are tracked with LFS
-        repo.lfs_track(["*.zip"])
-        repo.lfs_track(["*.pt"])
-        repo.push_to_hub(commit_message="Update tracked files", clean_ok=True)
-
-        dst = os.path.join(repo_local_path, "models")
-        src = os.path.join(input_dir, model)
-        if not os.path.exists(src):
+        repo_url = api.create_repo(repo_id=repo_id, exist_ok=True)
+
+        # Find src folder
+        src = Path(input_dir) / model
+        if not src.exists(src):
             if not input_dir:
                 raise FileNotFoundError(f"Could not find models under {src}.  Perhaps you forgot to set --input_dir?")
             else:
                 raise FileNotFoundError(f"Could not find models under {src}")
-        copytree(src, dst)
-
-        # Create the model card
-        write_model_card(repo_local_path, model)
-
-        # Push the model
-        # note: the error of not having anything to push will hopefully
-        # never happen since the README is updated to the millisecond
-        print("Pushing files to the Hub")
-        repo.push_to_hub(commit_message=f"Add model {args.version}")
-
-        tag = "v" + args.version
-        if repo.tag_exists(tag, remote=repo_url):
-            repo.delete_tag(tag, remote=repo_url)
-        repo.add_tag(tag_name=tag, message=f"Adding new version of models {tag}", remote=repo_url)
-        print(f"Added a tag for the new models: {tag}")
 
+        # Update model card in it
+        (src / "README.md).write_text(get_model_card(model))
+
+        # Upload model + model card
+        api.upload_folder(repo_id=repo_id, folder_path=src, version=f"Add model {args.version}")
```

Comment:
not `version`?  will have to check the API

Reply:
should this just still be `commit_message`?

Reply:
is there a convenient way to push everything to a subdirectory?  our old layout was, for example, to have a models directory under the root directory, such as here:

https://huggingface.co/stanfordnlp/stanza-es/tree/main
models, .gitattributes, README.md

all the subdirectories are now under the main directory:

https://huggingface.co/stanfordnlp/stanza-mr/tree/main
backward_charlm, depparse, etc etc, default.zip, .gitattributes, README.md

probably the `path_in_repo` flag?  although then there will need to be two separate pushes, one for the models and one for the README.  alternatively we could either change the code that arranges the models into their current layout, which is probably better, or we could change the code that downloads the models, and i'm kinda too lazy to do that.  but i believe rearranging the models when building up the model distribution is not going to be very difficult

Reply:
Oops yes it must be! `version` is not even a valid arg :see_no_evil: 

```suggestion
        api.upload_folder(repo_id=repo_id, folder_path=src, commit_message=f"Add model {args.version}")
```

Reply:
> probably the path_in_repo flag? although then there will need to be two separate pushes, one for the models and one for the README. alternatively we could either change the code that arranges the models into their current layout, which is probably better, or we could change the code that downloads the models, and i'm kinda too lazy to do that. but i believe rearranging the models when building up the model distribution is not going to be very difficult

Yes it is `path_in_repo` that you should use!
If you need 2 commits or to rearrange the dist structure locally, that's no problem (either way is good IMO). You're taking care of it IIUC?

Reply:
Yes, am mostly done!  Just need to verify that the new layout is happy - it takes a while to rearrange all the models, and there's never been a need for a model rearranging script that only does one language for testing purposes, so it just takes a little while